docs/modules/ROOT/pages/guides/snip12.adoc

```diff
@@ -101,16 +101,17 @@ In this case it can be computed as follows:
 
 [,cairo]
 ----
+// Since there's no u64 type in SNIP-12, we use u128 for `expiry` in the type hash generation.
 let message_type_hash = selector!(
-    "\"Message\"(\"recipient\":\"ContractAddress\",\"amount\":\"u256\",\"nonce\":\"felt\",\"expiry\":\"u64\")\"u256\"(\"low\":\"felt\",\"high\":\"felt\")"
+    "\"Message\"(\"recipient\":\"ContractAddress\",\"amount\":\"u256\",\"nonce\":\"felt\",\"expiry\":\"u128\")\"u256\"(\"low\":\"felt\",\"high\":\"felt\")"
 );
 ----
 
 which is the same as:
 
 [,cairo]
 ----
-let message_type_hash = 0x120ae1bdaf7c1e48349da94bb8dad27351ca115d6605ce345aee02d68d99ec1;
+let message_type_hash = 0xa2a7036c1f406af7c47722b209f23bd2f2d6ac21423c8c73bd92cf28409ee2;
```

Comment:
```suggestion
let message_type_hash = 0x1bd8318a34d524539867e6c110cb97d385c745575910b6c2afec623432ee55;
```

This is the hash I'm getting^

Can you confirm?

```
use core::to_byte_array::FormatAsByteArray;

let res = selector!("\"Message\"(\"recipient\":\"ContractAddress\",\"amount\":\"u256\",\"nonce\":\"felt\",\"expiry\":\"u128\")\"u256\"(\"low\":\"felt\",\"high\":\"felt\")");
println!("res: {}", res); // 49196983552140144393093620210836098680955957904181459224647480219440180821

let hex = res.format_as_byte_array(16);
println!("hex: 0x{}", hex); // 0x1bd8318a34d524539867e6c110cb97d385c745575910b6c2afec623432ee55
```

Reply:
Sorry! I was using `timestamp` for expiry and forgot to re-generate the hash.

The expected hash should be `0x28bf13f11bba405c77ce010d2781c5903cbed100f01f72fcff1664f98343eb6`, where `u256` is broken down to `u128` low, high values instead of `felt` ([ref](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md#when-x-is-a-u256)).


```
use core::to_byte_array::FormatAsByteArray;

let res = selector!("\"Message\"(\"recipient\":\"ContractAddress\",\"amount\":\"u256\",\"nonce\":\"felt\",\"expiry\":\"u128\")\"u256\"(\"low\":\"u128\",\"high\":\"u128\")");
println!("res: {}", res); // 1151882460384444625458237696336527865031165261341111302013491262776159780534

let hex = res.format_as_byte_array(16);
println!("hex: 0x{}", hex); // 0x28bf13f11bba405c77ce010d2781c5903cbed100f01f72fcff1664f98343eb6
```